File: Utility/Browser/HTML/modStripTags.vb
Imports HtmlAgilityPack
Imports System.Linq
Imports System.IO
Imports System.Collections.Generic
Imports System.Xml

Module modStripTags

    

    Public Enum partialM
        Yes = 1
        No = 2
    End Enum

    Sub StripTag( _
                  ByRef html As String, _
                  ByVal attribute As String, _
                  ByVal value As String, _
                  Optional ByVal partialMatch As partialM = partialM.No _
                )

        Dim doc As New HtmlDocument
        Dim outputFile As StringWriter

        outputFile = New StringWriter()

        html = CleanString(html)

        doc.LoadHtml(html)

        Dim nodesToRemove As List(Of HtmlNode) = Nothing

        nodesToRemove = doc.DocumentNode.Descendants() _
                                   .Where(Function(n) n.Attributes.Contains(attribute)) _
                                   .ToList()

        For Each node In nodesToRemove
            If node.Attributes(attribute).Value.Contains(value) Then
                node.Remove()
            End If
        Next

        doc.OptionOutputAsXml = True

        doc.Save(outputFile)

        html = outputFile.ToString

        outputFile.Close()

        doc = Nothing

    End Sub

    Sub StripTag( _
                  ByRef html As String, _
                  ByVal tag As String, _
                  Optional ByVal partialMatch As partialM = partialM.No _
                 )

        Dim doc As New HtmlDocument
        Dim outputFile As StringWriter

        outputFile = New StringWriter()

        html = CleanString(html)

        doc.LoadHtml(html)

        Dim nodesToRemove As List(Of HtmlNode) = Nothing

        Select Case partialMatch
            Case partialM.Yes
                nodesToRemove = doc.DocumentNode.Descendants() _
                                   .Where(Function(n) n.Name.Contains(tag)) _
                                   .ToList()
            Case partialM.No
                nodesToRemove = doc.DocumentNode.Descendants() _
                                   .Where(Function(n) n.Name = tag) _
                                   .ToList()
        End Select

        For Each node In nodesToRemove
            node.Remove()
        Next

        doc.OptionOutputAsXml = True

        doc.Save(outputFile)

        html = outputFile.ToString

        outputFile.Close()

        doc = Nothing

    End Sub

End Module

File: Downloader/HtmlScraper/Utility/Browser/HTML/modStripTags.cs
using HtmlAgilityPack;

namespace HtmlScraper.Utility.Browser.HTML
{

    static class modStripTags
    {



        public enum partialM
        {
            Yes = 1,
            No = 2
        }

        public static void StripTag(ref string html, string attribute, string value, partialM partialMatch = partialM.No)




        {

            var doc = new HtmlAgilityPack.HtmlDocument();
            StringWriter outputFile;

            outputFile = new StringWriter();

            html = modUtility.CleanString(html);

            doc.LoadHtml(html);

            List<HtmlNode> nodesToRemove = null;

            nodesToRemove = doc.DocumentNode.Descendants().Where(n => n.Attributes.Contains(attribute)).ToList();


            foreach (var node in nodesToRemove)
            {
                if (node.Attributes[attribute].Value.Contains(value))
                {
                    node.Remove();
                }
            }

            doc.OptionOutputAsXml = true;

            doc.Save(outputFile);

            html = outputFile.ToString();

            outputFile.Close();

            doc = null;

        }

        public static void StripTag(ref string html, string tag, partialM partialMatch = partialM.No)



        {

            var doc = new HtmlAgilityPack.HtmlDocument();
            StringWriter outputFile;

            outputFile = new StringWriter();

            html = modUtility.CleanString(html);

            doc.LoadHtml(html);

            List<HtmlNode> nodesToRemove = null;

            switch (partialMatch)
            {
                case partialM.Yes:
                    {
                        nodesToRemove = doc.DocumentNode.Descendants().Where(n => n.Name.Contains(tag)).ToList();

                        break;
                    }
                case partialM.No:
                    {
                        nodesToRemove = doc.DocumentNode.Descendants().Where(n => (n.Name ?? "") == (tag ?? "")).ToList();

                        break;
                    }
            }

            foreach (var node in nodesToRemove)
                node.Remove();

            doc.OptionOutputAsXml = true;

            doc.Save(outputFile);

            html = outputFile.ToString();

            outputFile.Close();

            doc = null;

        }

    }
}